File: benchmarks/async/01_sequential_chain.go
package main

import "fmt"

func step(n int) <-chan int {
	ch := make(chan int, 1)
	go func() { ch <- n }()
	return ch
}

func main() {
	total := 0
	for i := 1; i <= 100; i++ {
		total += <-step(i)
	}
	fmt.Printf("checksum = %d\n", total)
}

File: benchmarks/async/01_sequential_chain.py
"""Async benchmark 1: Sequential await chain (Python asyncio baseline)."""
import asyncio

async def step(n: int) -> int:
    return n

async def chain() -> int:
    total = 0
    for i in range(1, 101):
        total += await step(i)
    return total

def main():
    result = asyncio.run(chain())
    print(f"checksum = {result}")

if __name__ == "__main__":
    main()
